source/io/module_eggShell.js

```diff
@@ -260,15 +292,15 @@
             if (rank === 1) {
                 currArr = [];
                 for (i = 0; i < arrLength; i += 1) {
-                    currArr[i] = arr[startIndex + i];
+                    currArr[i] = arr[i];
                 }
                 return currArr;
             }
 
             // Perform nd array creation for rank > 1
             // TODO mraj this is O((m-1)n) for rank m. So rank 2 is O(n) and can be improved for rank > 2
-            currArr = arr;
-            var currStartIndex = startIndex;
+            currArr = Array.from(arr);
```

Comment:
The Array.from is going to make a full copy of the arr Array. Is making a copy of the array needed? Seems like it can be expensive

Reply:
You're right - I didn't need the Array.from here. I removed it.